### PR TITLE
add ssl_minimum_protocol and ssl_maximum_protocol for downstream

### PR DESCRIPTION
**Pull request by JLXIA**
Support ssl_minimum_protocol and ssl_maximum_protocol for downstream

### CODE REVIEW

File: src/go/util/transport_socket_test.go [hunk 1]
```diff
@@ -152,14 +155,18 @@ func TestCreateDownstreamTransportSocket(t *testing.T) {
...
 					}
 				}
 			} `,
 		},
 		{
-			desc:    "Downstream Transport Socket for TLS, for legacy ESPv1",
-			sslPath: "/etc/nginx/ssl",
+			desc:               "Downstream Transport Socket for TLS, for legacy ESPv1",
+			sslPath:            "/etc/nginx/ssl",
+			sslMaximumProtocol: "TLSv1.3",
```
Comment (qiwzhang) [inline src/go/util/transport_socket_test.go]: can we add a test to have both min and max?

Comment (JLXIA) [reply src/go/util/transport_socket_test.go]: Done

File: docker/generic/start_proxy.py [hunk 2]
```diff
@@ -540,6 +562,13 @@ def enforce_conflict_args(args):
     if len(port_flags) > 1:
         return "Multiple port flags {} are not allowed, please just use --listener_port flag".format(",".join(port_flags))
 
+    if args.ssl_minimum_protocol and args.ssl_minimum_protocol not in {"TLSv1.0", "TLSv1.1", "TLSv1.2", "TLSv1.3"}:
```
Comment (nareddyt) [inline docker/generic/start_proxy.py]: Suggest you use this instead: https://docs.python.org/3/library/argparse.html#choices

Comment (nareddyt) [reply docker/generic/start_proxy.py]: You can then also remove the values from the help text

Comment (JLXIA) [reply docker/generic/start_proxy.py]: Thanks, will do in following PR, since there are some other flags also need to change.

**Comment by nareddyt** [comment]
Please ignore the coverage job failure until #56 is merged

**Comment by JLXIA** [comment]
/ test all

**Pull request merged**